### PR TITLE
IGNITE-17578 Transactions: async cleanup processing on tx commit

### PR DESCRIPTION
**Pull request by denis-chudov**
https://issues.apache.org/jira/browse/IGNITE-17578

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionListener.java [hunk 1]
```diff
@@ -364,7 +365,8 @@ private void handleTxCleanupCommand(TxCleanupCommand cmd, long commandIndex, lon
 
         markFinished(txId, cmd.commit(), cmd.commitTimestamp(), cmd.txCoordinatorId());
 
-        Set<RowId> pendingRowIds = txsPendingRowIds.getOrDefault(txId, EMPTY_SET);
+        Set<RowId> rowIds = txsPendingRowIds.remove(txId);
```
[the remark this answers is not in the document]
Comment (denis-chudov) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionListener.java]: You are right, replacing `getOrDefault` with `remove` was not necessary here. Fixed.

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java [hunk 2]
```diff
@@ -1330,10 +1335,10 @@ private CompletableFuture<Void> processTxCleanupAction(TxCleanupReplicaRequest r
 
                         cleanupLocally(request.txId(), request.commit(), request.commitTimestamp());
 
-                        return raftClient
-                                .run(txCleanupCmd)
-                                .thenCompose(ignored -> allOffFuturesExceptionIgnored(txReadFutures, request)
-                                        .thenRun(() -> releaseTxLocks(request.txId())));
+                        raftClient.run(txCleanupCmd);
```
Comment (sanpwc) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java:1338]: At this point, while we don't have durable cleanup, it has sense to log an exception from raftClient.run()

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java [hunk 3]
```diff
@@ -1260,9 +1264,10 @@ private CompletableFuture<Object> finishTransaction(
     /**
      * Processes transaction cleanup request:
      * <ol>
-     *     <li>Run specific raft {@code TxCleanupCommand} command, that will convert all pending entries(writeIntents)
-     *     to either regular values({@link TxState#COMMITED}) or removing them ({@link TxState#ABORTED}).</li>
-     *     <li>Release all locks that were held on local Replica by given transaction.</li>
+     *     <li>Waits for finishing of transactional operations;</li>
```
[the remark this answers is not in the document]
Comment (denis-chudov) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java:1267]: yes

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java [hunk 4]
```diff
@@ -788,6 +790,8 @@ private CompletableFuture<?> createTablePartitionsLocally(
 
                 mvGc.addStorage(replicaGrpId, partitionUpdateHandlers.gcUpdateHandler);
 
+                Map<UUID, SortedSet<RowId>> txsPendingRowIds = new ConcurrentHashMap<>();
```
[the remark this answers is not in the document]
Comment (denis-chudov) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:793]: fixed

**Review by sanpwc — APPROVED** [review]

**Pull request merged**